File: 0977-squares-of-a-sorted-array/0977-squares-of-a-sorted-array.cpp
class Solution {
public:
    vector<int> sortedSquares(vector<int>& nums) {
        int n=nums.size();
        vector<int> ans(n,0);
        int start=0;
        int end=n-1;
        int k=n-1;
        while(start<=end){
            if(abs(nums[start]) >abs(nums[end]))
               {
                   ans[k]=nums[start]*nums[start];
                   start++;
               }
               else
               {
                  ans[k]=nums[end]*nums[end];
                   end--; 
               }
               k--;
        }
        return ans;       
    }
};

File: 0977-squares-of-a-sorted-array/0977-squares-of-a-sorted-array.java
class Solution {
    public int[] sortedSquares(int[] nums) {
        int s=nums.length;
        int result[] =new int[s];

        int i=0;
        int j=s-1;
        int k=s-1;

        while(i<=j){
            if(Math.abs(nums[i])<Math.abs(nums[j])){
                result[k]= nums[j]*nums[j];
                j--;
                k--;
            }
            else{
                result[k]= nums[i]*nums[i];
                i++;
                k--;
            }
        }

        return result;
    }
}